core/modules/system/system.menu.inc

```diff
@@ -0,0 +1,332 @@
+<?php
+
+/**
+ * Implements hook_block_configure().
+ */
+function _system_block_configure($delta = '', $settings = array()) {
+  // Create a pseudo form state.
+  // Merge in the default configuration.
+  $settings += system_menu_default_config();
+  $settings['delta'] = $delta;
+  return system_menu_configure_form($settings);
+}
+
+/**
+ * The default menu block configuration.
+ *
+ * @return array
+ */
+function system_menu_default_config() {
+  return array(
+    'parent' => 'main-menu:0',
+    'title_link' => 0,
+    'admin_title' => '',
+    'level' => 1,
+    'follow' => 0,
+    'depth' => 0,
+    'expanded' => 0,
+    'sort' => 0,
+    'menu_name' => 'main-menu',
+    'parent_mlid' => 0,
+  );
+}
+
+/**
+ * Returns the configuration form for a menu tree.
+ *
+ * @param $form_state
+ *   array An associated array of configuration options should be present in the
+ *   'values' key. If none are given, default configuration is assumed.
+ * @return
+ *   array The form in Form API format.
+ */
+function system_menu_configure_form($config) {
+  // Get the config from the form state.
+  if (!empty($config)) {
+    if (!empty($config['parent'])) {
+      list($config['menu_name'], $config['parent_mlid']) = explode(':', $config['parent']);
+    }
+  }
+
+  // Build the standard form.
+  //$form['#attached']['js'][] = backdrop_get_path('module', 'system') . '/menu-block.js';
+
+  $form['menu_name'] = array(
+    '#type' => 'value',
+    '#value' => $config['delta'],
+  );
+  $form['level'] = array(
+    '#type' => 'select',
+    '#title' => t('Starting level'),
+    '#default_value' => $config['level'],
+    '#options' => array(
+      '1' => t('1st level (primary)'),
+      '2' => t('2nd level (secondary)'),
+      '3' => t('3rd level (tertiary)'),
+      '4' => t('4th level'),
+      '5' => t('5th level'),
+      '6' => t('6th level'),
+      '7' => t('7th level'),
+      '8' => t('8th level'),
+      '9' => t('9th level'),
+    ),
+    '#description' => t('Blocks that start with the 1st level will always be visible. Blocks that start with the 2nd level or deeper will only be visible when the trail to the active menu item passes though the block’s starting level.'),
+  );
+  $form['depth'] = array(
+    '#type' => 'select',
+    '#title' => t('Maximum depth'),
+    '#default_value' => $config['depth'],
+    '#options' => array(
+      '1' => '1',
+      '2' => '2',
+      '3' => '3',
+      '4' => '4',
+      '5' => '5',
+      '6' => '6',
+      '7' => '7',
+      '8' => '8',
+      '9' => '9',
+      '0' => t('Unlimited'),
+    ),
+    '#description' => t('From the starting level, specify the maximum depth of the menu tree.'),
+  );
+
+  $form['#validate'][] = 'node_search_validatev';
```

Comment:
Looks like this should be removed or fixed? `node_search_validatev` probably isn't the right validation here.